airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/check/CheckOperation.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.check
+
+import io.airbyte.cdk.Operation
+import io.airbyte.cdk.output.ExceptionHandler
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Singleton
+
+@Singleton
+@Requires(property = Operation.PROPERTY, value = "check")
+@Requires(env = ["destination"])
+class CheckOperation(
```

Comment:
any reason why we wouldn't want an abstract class here, with the check() as an abstract function

Reply:
The `@Requires` aren't inherited. It would force each implementor to specify the conditions under which their implementation is run.